### PR TITLE
Add span data race tests to tracer test harness

### PR DESCRIPTION
**Pull request by iredelmeier**

### CODE REVIEW

**Comment by iredelmeier** [comment]
This is failing due to #230. #231 should fix this failure.

File: api/testharness/harness.go [hunk 1]
```diff
@@ -285,6 +290,122 @@ func (h *Harness) TestTracer(subjectFactory func() trace.Tracer) {
...
+			span.SetStatus(codes.Internal)
+		},
+		"#SetName": func(span trace.Span) {
+			span.SetName("new name")
+		},
+		"#SetAttribute": func(span trace.Span) {
+			span.SetAttribute(core.Key("key").String("value"))
+		},
+		"#SetAttributes": func(span trace.Span) {
+			span.SetAttributes(core.Key("key1").String("value"), core.Key("key2").Int(123))
```
Comment (jmacd) [inline api/testharness/harness.go]: Unrelated nit:  you can now write `key.String("key1", "value")` instead of `core.Key("key1").String("value")`.

Comment (krnowak) [reply api/testharness/harness.go]: That's from the "go.opentelemetry.io/api/key" package, just for the record.

File: api/testharness/harness.go [hunk 2]
```diff
@@ -285,6 +290,122 @@ func (h *Harness) TestTracer(subjectFactory func() trace.Tracer) {
...
+	for mechanismName, mechanism := range mechanisms {
+		h.t.Run(mechanismName, func(t *testing.T) {
+			for methodName, method := range methods {
+				t.Run(methodName, func(t *testing.T) {
+					t.Run("is thread-safe", func(t *testing.T) {
+						t.Parallel()
+
+						span := mechanism()
+
+						wg := &sync.WaitGroup{}
```
Comment (krnowak) [inline api/testharness/harness.go]: This does not need to be a pointer. A value is fine too.

File: api/testharness/harness.go [hunk 3]
```diff
@@ -285,6 +290,122 @@ func (h *Harness) TestTracer(subjectFactory func() trace.Tracer) {
...
+				actualCtx = ctx
+
+				return nil
+			})
+
+			return trace.CurrentSpan(actualCtx)
+		},
+	}
+
+	for mechanismName, mechanism := range mechanisms {
```
Comment (krnowak) [inline api/testharness/harness.go]: Should this be ran several times, not just once for each combination of mechanism and method?

Comment (jmacd) [reply api/testharness/harness.go]: I suppose someone could run with `-count N` to repeat the test?

Comment (iredelmeier) [reply api/testharness/harness.go]: I don't think it needs to be, given that `-race` should be sufficient?

**Pull request merged**